lib/parcel-request-handler.js

```diff
@@ -68,70 +71,75 @@ exports.getRegisteredAddresses = function (req, res){
+            res.json(parcels);
+          }
+        });
     } else {
       return res.status(404).end('invalid user');
     }
   });
-
-  new Parcel({'user_id': user_id})
-    .fetchAll()
-    .then(function(parcels) {
-      if (!parcels) {
-        res.status(200).end('no parcels attacked to this user');
-      } else {
-        res.json(parcels);
-      }
-    });
 };
 
 exports.togglePermissions = function (req, res) {
   var parcel_gid = req.body.parcel_gid;
   var restriction_start_time = req.body.restriction_start_time;
   var restriction_end_time = req.body.restriction_end_time;
-
+  console.log(parcel_gid);
   new Parcel({'parcel_gid': parcel_gid}).fetch().then(function(parcel){
       if (!parcel) {
+
         res.status(404).end('Failure to find parcel');
       } else {
-          planner.togglePermissions(parcel_gid, {
-            restriction_start_time: restriction_start_time,
-            restriction_end_time: restriction_end_time
-          });
-          parcel.customer.set({restriction_start_time: restriction_start_time,
-                               restriction_end_time: restriction_end_time}).then(function () {
-              res.json(parcel)
-          })       
+          parcel.set({restriction_start_time: restriction_start_time,
+                            restriction_end_time: restriction_end_time});
+                             return res.json(parcel);
+          // planner.togglePermissions(parcel_gid, {
+          //   restriction_start_time: restriction_start_time,
+          //   restriction_end_time: restriction_end_time
+          // }, function (success) {
+          //   parcel.set({restriction_start_time: restriction_start_time,
+          //                   restriction_end_time: restriction_end_time})
+          //                   .then(function () {res.json(parcel)});
+          // })       
         }
   });
 };
 
+//[ToDo: V2 - flight exceptions for dronePath]
 exports.setException = function (req, res) {
   var parcel_gid = req.body.parcel_gid;
   var exception_start_time = req.body.exception_start_time;
   var exception_end_time = req.body.exception_end_time;
   var drone_id = req.body.drone_id;
-  var exception_id = planner.setException(req.body)
 
 
-  new Exception({'exception_id': exception_id})
-    .fetch()
-    .then(function(exception) {
-      if (!exception) {
-        var newException = new ParcelData({
-          exception_id: exception_id,
-          drone_id: drone_id,
-          parcel_gid: parcel_gid,
-          exception_start_time: exception_start_time,
-          exception_end_time: exception_end_time
-        })
-        newException.save().then(function(newException) {
-            res.json(newException);
-            ExceptionCollection.add(newException);
-        });
-      } else {
-        res.status(404).end('invalid exception request');
-      }
-    });
+  planner.setException(req.body, function (exception_id) {  
+    new Exception({'exception_id': exception_id})
```

Comment:
will work with Arthur on this - it is restriction Exceptions. On the waffle to do list